test/middleware-mockserver.cjs

```diff
@@ -83,9 +83,7 @@ const mocks = {
 		if ( resp.set ) {
 			resp.set( headers );
 		} else {
-			for ( const key in headers ) {
-				resp.writeHead( 200, { [ key ]: headers[ key ] } );
-			}
+			resp.writeHead( 200, headers );
```

Comment:
@timmywil What's the reason for the above `resp.set` check? Is it to cater to Express as well? If that's the case, why wasn't this needed in other functions using `writeHead`? I see this has been added in #5318.

Reply:
Indeed it was to cater to Express, but I don't remember exactly why the others didn't need it. But, now that I think about it, it could be that I was seeing `writeHead` wipe out previous calls and I compensated by using an express method. You *may* be able to remove the `.set` and `.append` calls, but you can double-check with `node test/server.js` and `http://localhost:3000/test/`.

Reply:
The `.append(` call needs to stay due to different handling of duplicate header names but the `.set(` one is redundant now; removed.